testSrc/unit/io/flutter/utils/StdoutJsonParserTest.java

```diff
@@ -49,6 +51,21 @@ public void appendsWithoutLineBreaks() {
     );
   }
 
+  @Test
+  public void jsonWithLineBreaks() {
+    final StdoutJsonParser parser = new StdoutJsonParser();
+    parser.appendOutput("[{'foo':'bar'}]\ntest\n");
```

Comment:
somewhat minor: do we have to worry about `\r` or `\n` in the middle of a json block? I don't know off the top of my head how the dart json encoder deals with them.

Reply:
It seems this is dealt with fine, I tried putting \r, \n and \r\n in the middle of flutter tools output and it's recognized correctly.